File: current_session/70.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 3) return n;
        int a = 2;      // n-2
        int b = 3;      // n-1
        int cur = 3;
        while (cur != n) {
            int tmp = b;
            b = a+b;
            a = tmp;
            cur++;
        }
        return b;
    }
}

File: current_session/c++/70.cpp
#include <iostream>
#include <vector>

using namespace std;

// use dp
// dp[i] is the solution at n=i, where
// dp[i] = dp[i-1] + dp[i-2]
class Solution {
 public:
  int climbStairs(int n) {
    if (n <= 3) return n;
    int dp = 3, prev = 2, i = 3;
    while (i < n) {int tmp=dp; dp=dp+prev; prev=tmp; i++;}
    return dp;
  }
};